Return JuMP model after solving the basecase OPF

nychiang opened the pull request:
This PR creates a JuMP model and returns the object for user to modify, e.g., it can be used for TS-ACOPF where one needs to add TSI constraints into the OPF model.

nychiang committed:
create branch; return basecase model after solving it

cnpetra commented:
I understand the role of `solve_basecase_from_model` based on our in-person conversation. But it looks like there is no function to create a basecase model without first solving it. I would refactor `solve_basecase` and move the model creation code into a `create_basecase_model`. This way we can instrument the model with TSI constraint(s) without needing to solve it.

nychiang commented:
This should be in draft mode and I just changed it.
`solve_basecase_from_model` is the function that takes JuMP model as an input and solves it, i.e., the ideal step after `create_basecase_model`:
create_basecase_model --> create an JuMP model for basecase
solve_basecase_from_model --> solve input model for basecase

I have implemented the function `create_basecase_model`,  but currently it doesn't work with `tsslope`.
The issue here is that we pass values of some variables to python, before JuMP actually creates them. (@variable creates an AbstractVector, and concretize it later.) 

Right now the flowchart is:
solve_basecase --> create and solve the OPF problem for basecase
add TSI constraints to the model using tsslope
solve_basecase_from_model --> solves the new problem.

nychiang committed:
add create_basecase_model

cnpetra commented:
>> The issue here is that we pass values of some variables to python, before JuMP actually creates them. (@variable creates an AbstractVector, and concretize it later.)
I see. Maybe we can delay passing values to Python ML code and only send "abstract" or meta info?

cnpetra committed:
Merge pull request #1 from LLNL/EF_SCACOPF

Extended SCACOPF

Commit:
Creates the files required by ExaJuGo from MATLAB m file

Commit:
Update README file

Commit:
README file now informers the person where they can get wildfire dependent con files

Commit:
SCACOPF and ACOPF now can give optional directory to save information from the system.

Commit:
Updated README file.

Commit:
Updated README.md and added comments to SCACOPF and ACOPF

Commit:
Made small corrections

nychiang committed:
add PyCall

Commit:
Inform user add the additional command to save system information

Commit:
Removed unnecessary code

Commit:
Minor update

Commit:
Added comments to codes

Commit:
Data is save to a subdirectory

Commit:
Adding the California data to ExaJuGO example

Commit:
Move California system to ExaJuGO

Commit:
Updated README to account for moving the California example. More edits are needed that talk about save2psse.

Commit:
Added copyrights and license to both codes

Commit:
Added TODO item on README file.

Commit:
Removed repeated copyright notice.

cnpetra committed:
Merge pull request #8 from LLNL/Add_m_converter

Add m converter

nychiang committed:
fix conflicts due to the recent developments

Irabiel committed:
Added the load object to go structure

Irabiel committed:
Adding changes needed for TS project

Irabiel committed:
Made small update for basecase model to work with the mixed MOI.

nychiang committed:
write load in solution writer

Irabiel committed:
Cleaned up code.

Irabiel committed:
Mixed MOI can now write the solution.
